packages/web3-eth-accounts/src/account.ts

```diff
@@ -1,22 +1,173 @@
 import { utils, getPublicKey } from 'ethereum-cryptography/secp256k1';
+import { keccak256 } from 'ethereum-cryptography/keccak';
+import {
+	TransactionFactory,
+	FeeMarketEIP1559TxData,
+	AccessListEIP2930TxData,
+	TxData,
+} from '@ethereumjs/tx';
+import { ecdsaSign, ecdsaRecover } from 'secp256k1';
 import {
 	toChecksumAddress,
 	bytesToHex,
 	sha3Raw,
 	HexString,
 	isBuffer,
 	isValidString,
+	Address,
+	isHexStrict,
+	utf8ToHex,
+	hexToBytes,
 } from 'web3-utils';
-import { InvalidPrivateKeyError, PrivateKeyLengthError } from 'web3-common';
+import {
+	InvalidPrivateKeyError,
+	PrivateKeyLengthError,
+	UndefinedRAWTransactionError,
+	SignerError,
+	InvalidSignatureError,
+} from 'web3-common';
+import {
+	signatureObject,
+	signFunction,
+	signResult,
+	signTransactionFunction,
+	signTransactionResult,
+} from './types';
 
 // TODO Will be added later
 export const encrypt = (): boolean => true;
 
-// TODO Will be added later
-export const sign = (): boolean => true;
+/**
+ * Hashes the given message. The data will be UTF-8 HEX decoded and enveloped as follows: "\x19Ethereum Signed Message:\n" + message.length + message and hashed using keccak256.
+ */
 
-// TODO Will be added later
-export const signTransaction = (): boolean => true;
+export const hashMessage = (message: string): string => {
+	const messageHex = isHexStrict(message) ? message : utf8ToHex(message);
+
+	const messageBytes = hexToBytes(messageHex);
+
+	const preamble = `\x19Ethereum Signed Message:\n${messageBytes.length}`;
+
+	const ethMessage = Buffer.concat([Buffer.from(preamble), Buffer.from(messageBytes)]);
+
+	return `0x${Buffer.from(keccak256(ethMessage)).toString('hex')}`;
+};
+
+/**
+ * Signs arbitrary data. The value passed as the data parameter will be UTF-8 HEX decoded and wrapped as follows: "\x19Ethereum Signed Message:\n" + message.length + message
+ */
+export const sign = (data: string, privateKey: string): signResult => {
```

Comment:
Validation for `data` and `privateKey` should be added, additionally I think we should support `buffer`s are input for `privateKey` since we support it in `web3-eth-accounts`

Reply:
we dnt have private key as buffer input in 1.x https://web3js.readthedocs.io/en/v1.5.2/web3-eth-accounts.html#sign

Reply:
data is validated inside `hashMessage()`

Reply:
I understand, but to remain consistent with `web3-eth-accounts`, I'm suggesting to add support for buffers since it's a trivial addition

Aside from this, `privateKey` should be using the type `HexString` to match the rest of library where hex strings are expected and not arbitrary strings